bioagents/msa/msa.py

```diff
@@ -979,9 +981,11 @@ def _make_processor(self):
             directed_stmts = [s for s in new_processor.statements if
                               not isinstance(s, Complex)]
             new_processor.statements = directed_stmts
+
             # Look for new agents.
             for other_ag, stmt in self._iter_stmts(directed_stmts):
-                if other_ag is None or 'HGNC' not in other_ag.db_refs:
+                if other_ag is None or 'HGNC' not in other_ag.db_refs \
+                        or other_ag.name == ag.name:
```

Comment:
Similarly to the above, this change has been sitting in my local repo for a while. It looks like agents with shared names were an issue at one point, but again a review of this is warranted.

Reply:
I think this change is good, it's a simple way to eliminate adding upstream/downstreams that are the result of self-loop statements.